pkg/virt-controller/watch/migration.go

```diff
@@ -1020,11 +1020,26 @@ func (c *MigrationController) sync(key string, migration *virtv1.VirtualMachineI
 				return nil
 			}
 			if c.clusterConfig.NonRootEnabled() && util.CanBeNonRoot(vmi) == nil {
+				patches := []string{}
+
 				if vmi.Status.RuntimeUser != 107 {
-					patch := fmt.Sprintf(`[{ "op": "replace", "path": "/status/runtimeUser", "value": %s }]`, "107")
+					patches = append(patches, fmt.Sprintf(`{ "op": "replace", "path": "/status/runtimeUser", "value": %s }`, "107"))
+				}
+
+				// This is required in order to be able to update from v0.43-v0.51 to v0.52+
+				if vmi.Annotations == nil {
+					patches = append(patches, fmt.Sprintf(`{ "op": "add", "path": "/metadata/annotations", "value":  { "%s": "true"} }`, virtv1.DeprecatedNonRootVMIAnnotation))
+				} else if _, ok := vmi.Annotations[virtv1.DeprecatedNonRootVMIAnnotation]; !ok {
+					patches = append(patches, fmt.Sprintf(`{ "op": "add", "path": "/metadata/annotations/%s", "value": "true"}`, controller.EscapeJSONPointer(virtv1.DeprecatedNonRootVMIAnnotation)))
+
+				}
+
+				if len(patches) != 0 {
+					patch := fmt.Sprintf("[%s]", strings.Join(patches, ","))
```

Comment:
You can use the `GeneratePatchBytes` function here https://github.com/kubevirt/kubevirt/blob/main/pkg/controller/controller.go#L310.